synapse/lib/storm.py

```diff
@@ -3508,3 +3509,84 @@ async def execStormCmd(self, runt, genr):
 
                 await self.delEdges(node, verb, n2)
                 yield node, path
+
+class TagPruneCmd(Cmd):
+    '''
+    Prune a tag (or tags) from nodes.
+
+    This command will delete the tags specified as parameters from incoming nodes,
+    as well as all of their parent tags that don't have other tags as children.
+
+    For example, given a node with the tags:
+
+        #parent
+        #parent.child
+        #parent.child.grandchild
+
+    Pruning the parent.child.grandchild tag would remove all tags. If the node had
+    the tags:
+
+        #parent
+        #parent.child
+        #parent.child.step
+        #parent.child.grandchild
+
+    Pruning the parent.child.grandchild tag will only remove the parent.child.grandchild
+    tag as the parent tags still have other children.
+
+    Examples:
+
+        # Prune the parent.child.grandchild tag
+        inet:ipv4=1.2.3.4 | tag.prune parent.child.grandchild
+    '''
+    name = 'tag.prune'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('tags', default=[], nargs='*', help='Names of tags to prune.')
+        return pars
+
+    def hasChildTags(self, node, tag):
+        pref = tag + '.'
+        for ntag in node.tags:
+            if ntag.startswith(pref):
+                return True
+        return False
+
+    async def execStormCmd(self, runt, genr):
+
+        if self.runtsafe:
```

Comment:
need to check for perms to allow removing all parent tags here ( before beginning edits )